meson.build

```diff
@@ -401,6 +401,12 @@ if cc.get_id() == 'clang'
         ]
 endif
 
+if get_option('mode') == 'release'
+        possible_common_cc_flags += '-ftrivial-auto-var-init=zero'
+elif not get_option('valgrind') and get_option('b_sanitize') == 'none'
```

Comment:
I have to admit I'm not sure how to detect whether the sanitizers are used or not. `b_sanitize` is kind of broken (https://github.com/mesonbuild/meson/pull/9825) so it isn't always used to pass `-fsanitize=`. On OSS-Fuzz everything is passed via `CFLAGS/LDFLAGS`.

`valgrind` should be easy because `-Dvalgrind=true` is supposed to be passed to make `systemd` compatible with it.